src/Mod/Path/Generators/rotation_generator.py

```diff
@@ -0,0 +1,205 @@
+    norm = vec * 1  # copy vec so we don't alter original
+
+    if ref == refAxis.x:
+        plane = refAxis.y.value
+    elif ref == refAxis.y:
+        plane = refAxis.z.value
+    else:
+        plane = refAxis.x.value
+
+    norm.projectToPlane(FreeCAD.Vector(0, 0, 0), plane)
+
+    ref = ref.value
+    rot = FreeCAD.Rotation(norm, ref)
+    ang = math.degrees(rot.Angle)
+    angle = ang * plane.dot(rot.Axis)
+    PathLog.debug("relative ang: {}".format(angle))
+
+    return angle
+
+
+def __getCRotation(normalVector, cMin=-360, cMax=360):
+    """
+    Calculate the valid C axis rotations component to align the normalVector
+    with either the +y or -y axis.
+    multiple poses may be possible.  Returns a list of all valid poses
+    """
+    PathLog.debug("normalVector: {} cMin: {} cMax: {}".format(normalVector, cMin, cMax))
+
+    angle = relAngle(normalVector, refAxis.y)
+
+    # Given an angle, there are four possibilities; rotating +- to each of the
+    # two axes +y and -y
+    candidates = [angle]
+    if angle == 0:
+        candidates.append(180)
+    elif angle == 180:
+        candidates.append(0)
+    elif angle >= 0:
+        candidates.append(angle - 180)
+        candidates.append(180 + angle)
+        candidates.append(angle - 360)
+    else:
+        candidates.append(angle + 180)
+        candidates.append(-180 + angle)
+        candidates.append(angle + 360)
+
+    # final results are candidates that don't violate rotation limits
+    results = [c for c in candidates if c >= cMin and c <= cMax]
+
+    return results
+
+
+def __getARotation(normalVector, aMin=-360, aMax=360):
+    """
+    Calculate the A axis rotation component.
+    Final rotation is always assumed to be around +X. The sign of the returned
+    value indicates direction of rotation.
+
+    Returns None if rotation violates min/max constraints
+    """
+
+    angle = relAngle(normalVector, refAxis.z)
+
+    # only return a result if it doesn't violate rotation constraints
+    if angle > aMin and angle <= aMax:
+        return angle
+    else:
+        return None
+
+
+def generate(normalVector, aMin=-360, aMax=360, cMin=-360, cMax=360, compound=False):
+    """
+    Generates Gcode rotation to align a vector (alignVector) with the positive Z axis.
+
+    It first rotates around the Z axis (C rotation)
+    to align the vector the positive Y axis. Then around the X axis
+    (A rotation).
+
+    The min and max arguments dictate the range of motion allowed rotation in the respective
+    axis.
+    Default assumes continous rotation.
+
+    returns a list of path commands for the shortest valid solution
+
+    if compound is False, axis moves will be broken out to individual commands
+
+    The normalVector input from a typical face (f) can be obtained like this:
+
+        u, v = f.ParameterRange[:2]
+        n = f.normalAt(u,v)
+        plm = obj.getGlobalPlacement()
+        rot = plm.Rotation
+        normalVector = rot.multVec(n
```

Comment:
If the expectation is that this method should be used to obtain the normal vector for a target face, and it is assumed this method might be used more than once in PathScripts to work with rotation, then I suggest making this set of instructions into a simple @classmethod here as a helper/utility method available for usage.

Reply:
That's a good idea.  I hadn't thought of a class method but was considering some other options.  I don't know if it belongs in the generator module or a PathUtils or somewhere else.  I held off because I found the described solution didn't work in some cases and I didn't have time to track it down.